lua/bufferline/buffer.lua

```diff
@@ -64,34 +53,34 @@ local function get_name(opts, number)
     local ext_index = string.find(string.reverse(name), '%.')
 
     if ext_index ~= nil and (ext_index < max_len - #ellipsis) then
-      local extension = string.sub(name, -ext_index)
-      name = string.sub(name, 1, max_len - #ellipsis - #extension) .. ellipsis .. extension
+      local extension = substring(name, -ext_index)
+      name = substring(name, 1, max_len - #ellipsis - #extension) .. ellipsis .. extension
     else
-      name = string.sub(name, 1, max_len - #ellipsis) .. ellipsis
+      name = substring(name, 1, max_len - #ellipsis) .. ellipsis
     end
 
     -- safety to prevent recursion in any future edge case
-    name = string.sub(name, 1, max_len)
+    name = substring(name, 1, max_len)
   end
 
   return name
 end
 
 local separator = package.config:sub(1,1)
-local function get_unique_name (first, second)
-  local first_parts  = split(first,  separator)
-  local second_parts = split(second, separator)
+local function get_unique_name(first, second)
+  local first_parts  = vim.split(first,  separator)
+  local second_parts = vim.split(second, separator)
 
   local length = 1
-  local first_result  = join(slice(first_parts, -length),  separator)
-  local second_result = join(slice(second_parts, -length), separator)
+  local first_result  = concat(list_slice(first_parts, -length),  separator)
```

Comment:
I preferred naming the `local` functions after their full counterparts, e.g. `table.concat` -> `concat` instead of `join` to make it more obvious which function specifically is being called.

Reply:
Also note `utils.slice` was removed entirely in favor of `vim.list_slice` and/or `string.sub` depending on the type being sliced, since we don't have to maintain those implementations

Reply:
Note: I recently changed it so we use `strcharpart` in the same places as before, but called directly instead of through `slice`. `string.sub` was not slicing the string at the end of multibye characters so I reverted that change